app/uk/gov/hmrc/fileupload/controllers/envelope.scala

```diff
@@ -47,20 +49,45 @@ object EnvelopeReport {
       expiryDate = envelope.expiryDate,
       status = Some(envelope.status.name),
       metadata = envelope.metadata,
+      constraints = envelope.constraints,
       destination = envelope.destination,
       application = envelope.application,
       files = fileReports
     )
   }
-
 }
 
 case class CreateEnvelopeRequest(callbackUrl: Option[String] = None,
                                  expiryDate: Option[DateTime] = None,
-                                 metadata: Option[JsObject] = None)
+                                 metadata: Option[JsObject] = None,
+                                 constraints: Option[Constraints] = None) {
+  def constraintsWithDefaultsIfNotProvided(constraints: Option[Constraints], defaultConstraints: DefaultEnvelopeConstraints) = {
+    val defaultMaxCapacity = defaultConstraints.maxNumFiles
+    val defaultMaxSize = defaultConstraints.maxSize
+    val defaultMaxSizePerItem = defaultConstraints.maxSizePerItem
+
+    Constraints(
+      maxNumFiles = Some(constraints.map(_.maxNumFiles.getOrElse(defaultMaxCapacity)).getOrElse(defaultMaxCapacity).toString.toInt),
+      maxSize = Some(constraints.map(_.maxSize.getOrElse(defaultMaxSize)).getOrElse(defaultMaxSize).toString),
+      maxSizePerItem = Some(constraints.map(_.maxSizePerItem.getOrElse(defaultMaxSizePerItem)).getOrElse(defaultMaxSizePerItem).toString)
+    )
+
+  }
+
+}
+
+case class Constraints(maxNumFiles: Option[Int] = None,
+                       maxSize: Option[String] = None,
+                       maxSizePerItem: Option[String] = None)
+
+case class DefaultEnvelopeConstraints(maxNumFiles: Int,
```

Comment:
Yes.. I'd imagine you set up the default values in the Constraint case class above, so difficult to see the need of an additional DafaultEnvelopeConstraints case class too, no ?